src/utils/dates.js

```diff
@@ -10,3 +10,30 @@ const ONE_DAY_IN_MILLISECONDS = 86400000;
 export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
+
+/**
+ * Calculates the number of days between two JS Date objects.
+ * @example
+ * // Returns the number of days that have passed between the two dates.
+ * const startDate = new Date("2023-09-15");
+ * const endDate = new Date("2023-09-24");
+ * getDaysBetweenDates(startDate, endDate); // Output is 9 (days)
+ * @param {Date} startDate - The starting date.
+ * @param {Date} endDate - The ending date.
+ * @returns {number} - The number of days between the start and end dates.
+ */
+export function getDaysBetweenDates(startDate, endDate) {
+	try {
+		if (startDate instanceof Date && endDate instanceof Date) {
+			const days_between_in_milliseconds =
+				endDate.getTime() - startDate.getTime();
+
+			// Using Math.round() to ensure that fractional days are rounded to the nearest whole day
+			return Math.round(days_between_in_milliseconds / ONE_DAY_IN_MILLISECONDS);
+		} else {
+			console.log('Not a date');
```

Comment:
non-critical suggestion: maybe we can replace console.log() with ```throw new Error(Not a date)``` or ```console.error(error)```, that way the warning message has more context

Reply:
Thank you for the suggestion, @shangguanwang!

The `console.log()` within the `else` statement would be for debugging purposes, as we are already checking if the argument types are instances of **Date**. What I can do for the `else` is add something like:

```
console.log(`Not a date\nstartDate is of type: ${typeof startDate}
\nendDate is of type: ${typeof endDate}`);
```

That way, we would have more information about the specific data types of the arguments being received.

Then, for the `catch`, I can use `console.error()` as you suggested. I hope that makes sense. And again, I truly appreciate your suggestion!